### PR TITLE
feat: Add Introducing the Devfile AI Assistant blog post

### PR DESCRIPTION
**Pull request by tolusha**
https://pr-check-55-che-blog.surge.sh/2024/06/26/@tolusha-devfile-ai-assistant.html

### CODE REVIEW

File: _posts/2024-06-05-devfile-ai-assistant.adoc [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+description: >-
+  The Devfile AI Assistant is an online tool that simplifies the creation of devfiles for developers.
+categories: []
+keywords: ['devfile', 'ChatGPT', 'AI']
+slug: /@tolusha/devfile-ai-assistant
+---
+
+== Introducing the Devfile AI Assistant
+
+In software development, it's important to be efficient. Whether you're an experienced developer or just starting, setting up your development environment can be complex and time-consuming. That's where the link:https://chatgpt.com/g/g-Bm20CP2Rp-devfile-assistant[Devfile AI Assistant] comes in. It's designed to help developers create devfiles easily without needing in-depth knowledge of devfile syntax or structure. It uses a link:https://devfile.io[devfile knowledge base] to help you craft a devfile, making sure your development environment is set up right.
```
Comment (dkwon17) [inline _posts/2024-06-05-devfile-ai-assistant.adoc:14]:
```suggestion
In software development, it's important to be efficient. Whether you're an experienced developer or just starting, setting up your development environment can be complex and time-consuming. That's where the link:https://chatgpt.com/g/g-Bm20CP2Rp-devfile-assistant[Devfile AI Assistant] comes in. It's designed to help developers create devfiles easily without needing in-depth knowledge of the devfile syntax or structure. It uses a link:https://devfile.io[devfile knowledge base] to help you craft a devfile, making sure your development environment is set up right.
```

File: _posts/2024-06-05-devfile-ai-assistant.adoc [hunk 2]
```diff
@@ -0,0 +1,322 @@
+---
```
Comment (ibuziuk) [inline _posts/2024-06-05-devfile-ai-assistant.adoc:1]: Let's change the publication date to the next Monday `2024-06-26`

**Comment by ibuziuk** [comment]
Great read! The only recommendation would be adding a `What is a devfile` section that would laconically explain what is it and how it is used in Eclipse Che:

File: _posts/2024-06-05-devfile-ai-assistant.adoc [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+layout: post
+author: Anatolii Bazko
+description: >-
+  The Devfile AI Assistant is an online tool that simplifies the creation of devfiles for developers.
+categories: []
+keywords: ['devfile', 'ChatGPT', 'AI']
+slug: /@tolusha/devfile-ai-assistant
+---
+
+== Introducing the Devfile AI Assistant
```
Comment (ibuziuk) [inline _posts/2024-06-05-devfile-ai-assistant.adoc:12]: I would add a `== What is a devfile?` section with a brief introduction to a devfile and how it is used in Eclipse Che:

Screenshot 2024-06-19 at 10 29 53

**Comment by l0rd** [comment]
👍 @tolusha. I like that. 

I have a few suggestions:
- Mention the [VS Code walkthrough extension](https://github.com/devfile/vscode-walkthrough-extension) that is less fancy but that does the job too.
- Avoid using as an example the `Add a project to the devfile`. [This controversial](https://github.com/eclipse-che/che/issues/21188). The suggestion is to add the Devfile to the project rather than hardcode the URL and branch in the Devfile.
- Add a prompt that pushes further ChatGPT capabilities so that it does a task that is usually tedious and difficult for users. Some ideas that come to my mind:
  - "Update the component image to build and test the source code in the repository <repository-url>"
  - "Write a Dockerfile to extend the universal developer image to include the development tool <tool-name> and write a GitHub workflow to build and push the image automatically" etc...

**Comment by tolusha** [comment]
Hello @l0rd 
I've mentioned the VSCode walk-through extension in the blog post.
With regards to the other remarks:
> Avoid using as an example the Add a project to the devfile. https://github.com/eclipse-che/che/issues/21188. The suggestion is to add the Devfile to the project rather than hardcode the URL and branch in the Devfile.

The idea is to show that Devfile AI Assistant can easily do any task related to a devfile.
So, if specification allows to do it, than it make sense to show off.

> Add a prompt that pushes further ChatGPT capabilities so that it does a task that is usually tedious and difficult for users. Some ideas that come to my mind:

Devfile AI Assistant is specifically configured by its instructions to help a developer to create the Devfile from the scratch. 
It utilizes devfile specification and other resources. So, the mentioned prompts are a bit out of context right now and can be used to improve the Assistant in the future.

**Comment by l0rd** [comment]
> The idea is to show that Devfile AI Assistant can easily do any task related to a devfile.
So, if specification allows to do it, than it makes sense to show off.

That's not a big issue but I think it's important to be on the same page about `projects`. The devfile is the configuration file of your project development environment. Its place is in your project git repository. The git repo URL is implicit.

**Review by l0rd — APPROVED** [review]

File: _posts/2024-06-26-devfile-ai-assistant.adoc [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+categories: []
+keywords: ['devfile', 'ChatGPT', 'AI']
+slug: /@tolusha/devfile-ai-assistant
+---
+
+== Introducing the Devfile AI Assistant
+
+In software development, it's important to be efficient. Whether you're an experienced developer or just starting, setting up your development environment can be complex and time-consuming. That's where the link:https://chatgpt.com/g/g-Bm20CP2Rp-devfile-assistant[Devfile AI Assistant] comes in. It's designed to help developers create devfiles easily without needing in-depth knowledge of the devfile syntax or structure. It uses a link:https://devfile.io[devfile knowledge base] to help you craft a devfile, making sure your development environment is set up right.
+
+Another tool that simplifies creating devfile is the link:https://github.com/devfile/vscode-walkthrough-extension[VS Code Walkthrough Extension for Devfile]. It provides an interactive guide to help users understand and utilize devfiles within Visual Studio Code. Using a wizard, you can easily create a devfile to customize your Cloud Development Environment.
```
Comment (ibuziuk) [inline _posts/2024-06-26-devfile-ai-assistant.adoc:16]:
```suggestion
NOTE: Another tool that simplifies creating devfile is the link:https://github.com/devfile/vscode-walkthrough-extension[VS Code Walkthrough Extension for Devfile]. It provides an interactive guide to help users understand and utilize devfiles within Visual Studio Code. Using a wizard, you can easily create a devfile to customize your Cloud Development Environment.
```

I would extract it as a  adoc`NOTE:` to make it more readable

File: _posts/2024-06-26-devfile-ai-assistant.adoc [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+
+
+== What is a devfile?
+
+Devfiles are configuration files that define development environments. They specify the components, commands, events etc. that make up a development workspace. They play a central role in the workflow of Eclipse Che.
+
+In Eclipse Che, devfiles are used to automate the setup of development environments. By providing a devfile, developers can ensure a consistent environment across the team, getting rid of the "works on my machine" problems. When a devfile is applied, Eclipse Che provisions the specified containers, clones the necessary repositories, and sets up the workspace with all the tools and dependencies required for development. It makes them an essential tool for modern cloud-based development.
```
Comment (ibuziuk) [inline _posts/2024-06-26-devfile-ai-assistant.adoc:23]:
```suggestion
In Eclipse Che, devfiles are used to automate the setup of development environments. By providing a devfile, developers can ensure a consistent environment across the team, getting rid of the "works on my machine" problems. When a devfile is applied, Eclipse Che provisions the specified containers, clones the necessary repositories, and sets up the workspace with all the tools and dependencies required for development. This makes them an essential tool for modern cloud-based development.
```

File: _posts/2024-06-26-devfile-ai-assistant.adoc [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+      image: quay.io/devfile/universal-developer-image:latest
+----
+
+. Add a project to the devfile.
++
+**Request:**
++
+[source]
+----
+Add https://github.com/eclipse-che/che-server git project.
```
Comment (ibuziuk) [inline _posts/2024-06-26-devfile-ai-assistant.adoc:59]: I agree with Mario, that project part is a bit controversial. I think we need to add a note about that

File: _posts/2024-06-26-devfile-ai-assistant.adoc [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+schemaVersion: 2.3.0
+metadata:
+  name: basic-devfile
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:latest
+----
+
+. Add a project to the devfile.
```
Comment (ibuziuk) [inline _posts/2024-06-26-devfile-ai-assistant.adoc:53]:
```suggestion
. Add a project to the devfile.

IMPORTANT: Even though it is possible to define projects in a devfile explicitly, it is recommended to add `.devfile.yaml` or `devfile.yaml` file to the root of the repository which allows skipping the project definition on the devfile level.
```

File: _posts/2024-06-26-devfile-ai-assistant.adoc (hunk 6 above)
Comment (ibuziuk) [reply _posts/2024-06-26-devfile-ai-assistant.adoc:59]: Suggestion - https://github.com/eclipse-che/blog/pull/55/files#r1648590270

**Review by ibuziuk — APPROVED** [review]
great article 👍 
Added a few minor improvements which hopefully make it even more awesome

**Pull request merged**